### PR TITLE
[pack-script] Re-construct of pack scirpts

### PR DESCRIPTION
**Pull request by haoyunfeix**
1. Separate the utils functions from pack.py to utils.py
    2. Separate the backends in pack.py to build_*.py

    BUGS=XWALK-5037 XWALK-5038 XWALK-5039

### CODE REVIEW

**Comment by haoyunfeix** [comment]
Did a try-run on full release with multi-process, all packages can be build successfully, include arm/x86 [,embedded/shared] [,apk/cordova/embeddingapi/wgt] [,ant/gradle/maven]

**Comment by jennycao** [comment]
LGTM

**Comment by jennycao** [comment]
@yugang @Honry @zqzhang @Shao-Feng 
Pls also help to review. The design for the pack script update is tracked at 
https://crosswalk-project.org/jira/browse/XWALK-5036

File: tools/build/utils.py [hunk 1]
```diff
@@ -0,0 +1,308 @@
...
+            msg = "\33[01m" + msg[1:] + "\033[0m"
+        elif msg[0] == "=":
+            msg = "\33[07m" + msg + "\033[0m"
+        levelname = record.levelname
+        if levelname in colors:
+            msg_color = "\033[0;%dm" % (
+                31 + colors[levelname]) + msg + "\033[0m"
+            record.msg = msg_color
+
+        return logging.Formatter.format(self, record)
```
Comment (Honry) [inline tools/build/utils.py]: Please add two white lines before each function.

File: tools/build/utils.py [hunk 2]
```diff
@@ -0,0 +1,308 @@
+#!/usr/bin/env python
+#
+# Copyright (c) 2014 Intel Corporation.
```
Comment (Honry) [inline tools/build/utils.py]: Please upgrade all the Copyright to 2015.

**Comment by haoyunfeix** [comment]
Formatted as below:
1. Change copyright from 2014 -> 2015
2. Add two while lines between each function (add one white line for functions in Class or in other function)

**Pull request merged**